src/robot/parsing/model.py

```diff
@@ -595,20 +595,33 @@ def __iter__(self):
 
 
 class ForLoop(_WithSteps):
+    """
+    Properties:
+
+    - flavor: The value of the 'IN' item, with spaces removed.
+              Typically 'IN', 'INRANGE', 'INZIP', 'INEUMERATE', etc.(?)
+    - range:  True if flavor is 'INRANGE'
+    - vars:   Variables set per-iteration by this loop
+    - items:  Items to loop over (or otherwise parameters, like with INRANGE)
+    - comment: a comment, or None.
+    - steps:  A list of steps in the loop.
+    """
 
     def __init__(self, declaration, comment=None):
-        self.range, index = self._get_range_and_index(declaration)
+        self.flavor, index = self._get_flavor_and_index(declaration)
+        # self.range is preserved for backwards-compatibility
+        self.range = self.flavor == 'INRANGE'
```

Comment:
Adding more generic `self.flavor` is a good idea. I hope we wouldn't need `self.range` anymore after that. I know removing it is a backwards incompatible change, but we are pretty deep in Robot internals here so such a change ought to be OK in a major release.